lib/pages/chat/send_file_dialog/send_file_dialog.dart

```diff
@@ -0,0 +1,204 @@
+import 'package:fluffychat/di/global/get_it_initializer.dart';
+import 'package:fluffychat/domain/usecase/send_files_on_web_with_caption_interactor.dart';
+import 'package:fluffychat/domain/usecase/send_media_on_web_with_caption_interactor.dart';
+import 'package:fluffychat/pages/chat/input_bar/focus_suggestion_controller.dart';
+import 'package:fluffychat/presentation/extensions/send_file_web_extension.dart';
+import 'package:fluffychat/presentation/list_notifier.dart';
+import 'package:fluffychat/utils/matrix_sdk_extensions/matrix_file_extension.dart';
+import 'send_file_dialog_view.dart';
+import 'package:fluffychat/presentation/enum/chat/send_media_with_caption_status_enum.dart';
+import 'package:flutter/material.dart';
+
+import 'package:matrix/matrix.dart';
+
+class SendFileDialog extends StatefulWidget {
+  final Room? room;
+  final List<MatrixFile> files;
+
+  const SendFileDialog({
+    this.room,
+    required this.files,
+    Key? key,
+  }) : super(key: key);
+
+  @override
+  SendFileDialogController createState() => SendFileDialogController();
+}
+
+class SendFileDialogController extends State<SendFileDialog> {
+  final sendMediaOnWebWithCaptionInteractor =
+      getIt.get<SendMediaOnWebWithCaptionInteractor>();
+
+  final sendFilesOnWebWithCaptionInteractor =
+      getIt.get<SendFilesOnWebWithCaptionInteractor>();
+
+  final focusSuggestionController = FocusSuggestionController();
+
+  final TextEditingController textEditingController = TextEditingController();
+
+  final FocusNode captionsFocusNode = FocusNode();
+
+  final ValueKey sendFileDialogTypeAheadKey =
+      const ValueKey('sendFileDialogTypeAhead');
+
+  bool isSendMediaWithCaption = true;
+
+  ListNotifier<MatrixFile> filesNotifier = ListNotifier([]);
+
+  Map<MatrixFile, MatrixImageFile?> thumbnails = {};
+
+  ValueNotifier<double> maxMediaSizeNotifier = ValueNotifier(double.infinity);
+
+  ValueNotifier<bool> haveErrorFilesNotifier = ValueNotifier(false);
+
+  @override
+  void initState() {
+    super.initState();
+    filesNotifier = ListNotifier(widget.files);
+    isSendMediaWithCaption = _isShowSendMediaDialog(
+      filesNotifier.value,
+      widget.room,
+    );
+    requestFocusCaptions();
+    WidgetsBinding.instance.addPostFrameCallback((timeStamp) async {
+      await loadMaxMediaSize();
+      await loadThumbnailsForMedia();
+    });
+  }
+
+  @override
+  void dispose() {
+    textEditingController.dispose();
+    super.dispose();
+  }
+
+  Future<void> loadMaxMediaSize() async {
+    if (widget.room == null) {
+      return;
+    }
+    final mediaConfig = await widget.room?.client.getConfig();
+    maxMediaSizeNotifier.value =
+        mediaConfig?.mUploadSize?.toDouble() ?? double.infinity;
```

Comment:
IMO: Create a default value size, not use `double.infinity`

Reply:
no need, because later we get media upload size from server config

Reply:
what happen if an error throw when you get config?

Reply:
okay, we should put it in a `try/catch`